pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -212,112 +207,110 @@ func (p *DefaultProvider) ensureLaunchTemplate(ctx context.Context, options *ami
 	if err != nil {
 		return nil, err
 	}
-	launchTemplateDataTags := []*ec2.LaunchTemplateTagSpecificationRequest{
-		{ResourceType: aws.String(ec2.ResourceTypeNetworkInterface), Tags: utils.MergeTags(options.Tags)},
+	launchTemplateDataTags := []ec2types.LaunchTemplateTagSpecificationRequest{
+		{ResourceType: ec2types.ResourceTypeNetworkInterface, Tags: utils.MergeTags(options.Tags)},
 	}
-	// Add the spot-instances-request tag if trying to launch spot capacity
 	if options.CapacityType == karpv1.CapacityTypeSpot {
-		launchTemplateDataTags = append(launchTemplateDataTags, &ec2.LaunchTemplateTagSpecificationRequest{ResourceType: aws.String(ec2.ResourceTypeSpotInstancesRequest), Tags: utils.MergeTags(options.Tags)})
+		launchTemplateDataTags = append(launchTemplateDataTags, ec2types.LaunchTemplateTagSpecificationRequest{ResourceType: ec2types.ResourceTypeSpotInstancesRequest, Tags: utils.MergeTags(options.Tags)})
 	}
 	networkInterfaces := p.generateNetworkInterfaces(options)
-	output, err := p.ec2api.CreateLaunchTemplateWithContext(ctx, &ec2.CreateLaunchTemplateInput{
+	output, err := p.ec2api.CreateLaunchTemplate(ctx, &ec2.CreateLaunchTemplateInput{
 		LaunchTemplateName: aws.String(LaunchTemplateName(options)),
-		LaunchTemplateData: &ec2.RequestLaunchTemplateData{
-			BlockDeviceMappings: p.blockDeviceMappings(options.BlockDeviceMappings),
-			IamInstanceProfile: &ec2.LaunchTemplateIamInstanceProfileSpecificationRequest{
+		LaunchTemplateData: &ec2types.RequestLaunchTemplateData{
+			BlockDeviceMappings: lo.FromSlicePtr(p.blockDeviceMappings(options.BlockDeviceMappings)),
+			IamInstanceProfile: &ec2types.LaunchTemplateIamInstanceProfileSpecificationRequest{
 				Name: aws.String(options.InstanceProfile),
 			},
-			Monitoring: &ec2.LaunchTemplatesMonitoringRequest{
+			Monitoring: &ec2types.LaunchTemplatesMonitoringRequest{
 				Enabled: aws.Bool(options.DetailedMonitoring),
 			},
 			// If the network interface is defined, the security groups are defined within it
-			SecurityGroupIds: lo.Ternary(networkInterfaces != nil, nil, lo.Map(options.SecurityGroups, func(s v1.SecurityGroup, _ int) *string { return aws.String(s.ID) })),
+			SecurityGroupIds: lo.Ternary(networkInterfaces != nil, nil, lo.Map(options.SecurityGroups, func(s v1.SecurityGroup, _ int) string { return s.ID })),
 			UserData:         aws.String(userData),
 			ImageId:          aws.String(options.AMIID),
-			MetadataOptions: &ec2.LaunchTemplateInstanceMetadataOptionsRequest{
-				HttpEndpoint:            options.MetadataOptions.HTTPEndpoint,
-				HttpProtocolIpv6:        options.MetadataOptions.HTTPProtocolIPv6,
-				HttpPutResponseHopLimit: options.MetadataOptions.HTTPPutResponseHopLimit,
-				HttpTokens:              options.MetadataOptions.HTTPTokens,
+			MetadataOptions: &ec2types.LaunchTemplateInstanceMetadataOptionsRequest{
+				HttpEndpoint:            ec2types.LaunchTemplateInstanceMetadataEndpointState(lo.FromPtr(options.MetadataOptions.HTTPEndpoint)),
```

Comment:
@engedaam and I were talking and we will do this in a separate PR. If we are going to do this we would want to do both this and change IOP and Throughput in BlockDecide to get rid of the ConvertFromNil function and this is already a really big PR so that cleanup and hash bump can happen later.